### PR TITLE
Gateway list dashboard and widgets

### PR DESCRIPTION
**Pull request by MrKartoshka**
## Pull Request description
New features regarding gateways

## General checklist

- [x] This PR has the corresponding CE PR or contains unique code changes related to PE only (does not require same changes in CE).
- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
  
## Front-End feature checklist

- [x] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [x] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [x] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-pe-ui-help)

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

Created new dashboard "Gateway" for showing gateweys user have access to.
Dashboard consists of

Gateway List table
image

Gateway state 
image

Gateway Logs (allso present for selected connector)
image

Gateway stayistics
image

Remote Shell
image

RPC state (allso present for selected connector)
image

Gateway configuration widget (newly created duplicates functionality provided by table cell action for newly created gateways
image

Gateway connectors widget (new dashboard state for it from gateway list's cell action)
image

Gateway List have cell action wich displays modal window with docker run commans for  WinOS or Linux same as launch command from gateway state.
image

Gateway List have cell action wich displays modal window with gateway configuration form.
image
configuration will be storred as shared attribute in device

Gateway List widget header action will create new device and after creation will redirect to configuration state.
image

### CODE REVIEW

**Comment by vvlladd28** [comment]
This PR does not support internationalization. It must be added.

File: ui-ngx/src/app/modules/home/components/device/device-credentials.module.ts [hunk 1]
```diff
@@ -22,14 +22,16 @@ import { DeviceCredentialsComponent } from '@home/components/device/device-crede
...
 
 @NgModule({
   declarations: [
     CopyDeviceCredentialsComponent,
     DeviceCredentialsComponent,
     DeviceCredentialsLwm2mComponent,
     DeviceCredentialsLwm2mServerComponent,
-    DeviceCredentialsMqttBasicComponent
+    DeviceCredentialsMqttBasicComponent,
+    DeviceExampleCommandComponent
```
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/components/device/device-credentials.module.ts:34]: This improvement work in other PR: https://github.com/thingsboard/thingsboard/pull/8803

File: ui-ngx/src/app/shared/components/file-input.component.scss [hunk 2]
```diff
@@ -93,4 +93,8 @@ $previewSize: 100px !default;
       padding: 0 16px;
     }
   }
+
+  .pe {
+    pointer-events: auto;
+  }
```
Comment (vvlladd28) [inline ui-ngx/src/app/shared/components/file-input.component.scss:99]: Incorrect class name

File: ui-ngx/src/app/shared/components/file-input.component.html [hunk 3]
```diff
@@ -16,7 +16,12 @@
 
 -->
 <div class="tb-container">
-  <label class="tb-title" [ngClass]="{'tb-required': !disabled && required}">{{ label }}</label>
+  <label class="tb-title" [ngClass]="{'tb-required': !disabled && required}">{{ label }}
+    <mat-icon *ngIf="hint" class="material-icons-outlined pe"
+              matSuffix style="cursor:pointer;"
+              matTooltip="{{hint}}">info_outlined
+    </mat-icon>
```
Comment (vvlladd28) [inline ui-ngx/src/app/shared/components/file-input.component.html:23]: matSuffix invalid attribute

File: ui-ngx/src/app/shared/components/dialog/gateway-remote-configuration-dialog.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  selector: 'tb-activation-link-dialog',
+  templateUrl: './gateway-remote-configuration-dialog.html'
+})
+
+export class GatewayRemoteConfigurationDialogComponent extends DialogComponent<GatewayRemoteConfigurationDialogComponent,
+  boolean> implements OnInit {
+
+  gatewayName: string;
+
+  gatewayForm: FormControl;
```
Comment (vvlladd28) [inline ui-ngx/src/app/shared/components/dialog/gateway-remote-configuration-dialog.ts:39]: Maybe gatewayControl?

**Review by vvlladd28 — CHANGES_REQUESTED** [review]

File: ui-ngx/src/app/modules/home/components/device/device-credentials.module.ts (hunk 1 above)
Comment (MrKartoshka) [reply ui-ngx/src/app/modules/home/components/device/device-credentials.module.ts:34]: Will use device api after it will be implemented

**Review by vvlladd28 — APPROVED** [review]

**Pull request merged**